node/node.go

```diff
@@ -0,0 +1,452 @@
+package node
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"go.sia.tech/jape"
+	"go.sia.tech/renterd/autopilot"
+	"go.sia.tech/renterd/bus"
+	"go.sia.tech/renterd/internal/consensus"
+	"go.sia.tech/renterd/internal/stores"
+	"go.sia.tech/renterd/wallet"
+	"go.sia.tech/renterd/worker"
+	"go.sia.tech/siad/modules"
+	mconsensus "go.sia.tech/siad/modules/consensus"
+	"go.sia.tech/siad/modules/gateway"
+	"go.sia.tech/siad/modules/transactionpool"
+	"go.sia.tech/siad/types"
+	"golang.org/x/crypto/blake2b"
+)
+
+type WorkerConfig struct {
```

Comment:
should we refactor this into

```
type NodeConfig struct {
    Autopilot struct {...}
    Bus struct {...}    
    Worker struct {...}
}
```

Reply:
I don't think so. There is really no point in having a config for the node if the parameters easily fit in the signature of `NewNode`. It just makes things more prone to errors.

The node can then decide what the best config is for each of the autopilot, bus and worker, create it and pass it on.